### PR TITLE
Fix content_blob contract drift across datastore schemas

### LINKED ISSUE #108: db9 content_blob contract is inconsistent between real schema, tests, and runtime assumptions
## Problem

The repository currently carries two conflicting assumptions about `files.content_blob` for the `db9` provider:

- the real DB9 schema defines `content_blob BYTEA`
- the datastore schema helper and schema tests assume `db9` does **not** have `content_blob`

This leaves the codebase with an inconsistent contract around whether `content_blob` should exist for DB9 tenants.

## Current evidence

Real DB9 schema includes `content_blob`:

- `pkg/tenant/schema_db9.go:38`

Datastore test helper assumes only TiDB providers have `content_blob`:

- `pkg/datastore/schema_test_helper_test.go:19`
- `pkg/datastore/schema_test_helper_test.go:24`

Schema test explicitly expects `db9` to have no `content_blob`:

- `pkg/datastore/schema_test.go:23`

Runtime datastore code supports both schemas via `hasContentBlob`:

- `pkg/datastore/store.go:128`
- `pkg/datastore/store.go:392`
- `pkg/datastore/store.go:418`
- `pkg/datastore/store.go:441`

But backend read paths still assume inline DB-backed files can read from `File.ContentBlob` when `StorageType == StorageDB9`:

- `pkg/backend/dat9.go:669`
- `pkg/backend/image_extract.go:317`

## Why this matters

Today this is partially masked because `tenant.SmallInDB()` returns `false` for `db9`, so DB9 tenants normally avoid the small-inline-file path.

However, the repository still has a broken contract:

- schema says DB9 has `content_blob`
- tests say DB9 does not
- datastore tries to support both
- backend code effectively assumes `ContentBlob` is present for direct DB-backed reads

If any DB9 runtime path ever stores/reads inline content while running against the no-blob contract, reads can silently return empty data because `scanFileNoBlob()` leaves `File.ContentBlob` unset.

## History

This appears to come from a sequence of commits:

1. `87dbd1a1046d7fce4751832c68bfa11682a48f42` introduced the provider-driven schema test helper and encoded the assumption that only `tidb_zero` / `tidb_cloud_starter` have `content_blob`.
2. `47efd7e2dc0711055f0b498898fe00de22e5f162` split schema init by provisioner and introduced `pkg/tenant/schema_db9.go` with `content_blob BYTEA` in the real DB9 schema.
3. `ff8f1b0cb859bf088a97930c37e2b7938465dac6` kept the real DB9 schema with `content_blob`, while also adding/retaining tests that still expect `db9` to have no `content_blob`.

## Suggested fix

Pick one contract and align the codebase around it:

Option A: `db9` should have `content_blob`

- update datastore schema helper/tests to expect `content_blob` for `db9`
- keep runtime behavior as-is
- add an end-to-end test that validates DB9 schema/runtime compatibility

Option B: `db9` should not have `content_blob`

- remove `content_blob` from `pkg/tenant/schema_db9.go`
- audit backend read/write paths so DB-backed reads do not rely on `File.ContentBlob`
- add explicit guards/tests to prevent silent empty reads

## Additional note

The current datastore schema test already has a TODO noting that it does not validate the real Postgres-backed DB9 schema end-to-end. That gap is likely why this drift survived.

### PR DESCRIPTION
**Pull request by JaySon-Huang**
## Summary
- align datastore schema tests and TiDB schema builders on the current contract that every supported provider includes `files.content_blob`
- remove legacy datastore read/write branches that tried to operate without `content_blob`, so runtime code matches the real schema
- document the `make test` Podman fallback for environments where direct `go test` fails with `rootless Docker not found`

## Testing
- `source ./scripts/test-podman.sh && go test -v ./pkg/datastore ./pkg/tenant`
- `source ./scripts/test-podman.sh && go test -v ./pkg/datastore ./pkg/backend ./pkg/server`

Closes #108

### CODE REVIEW

File: pkg/datastore/store.go [hunk 1]
```diff
@@ -610,11 +575,7 @@ func (s *Store) ListDir(ctx context.Context, parentPath string) (out []*NodeWith
 	defer observeStoreOp(ctx, "list_dir", start, &err)
 
 	q := `SELECT fn.node_id, fn.path, fn.parent_path, fn.name, fn.is_directory, fn.file_id, fn.created_at,
-		f.file_id, f.storage_type, f.storage_ref, `
-	if s.hasContentBlob {
-		q += `f.content_blob, `
-	}
-	q += `f.content_type, f.size_bytes,
+		f.file_id, f.storage_type, f.storage_ref, f.content_blob, f.content_type, f.size_bytes,
```
[the remark this answers is not in the document]
Comment (JaySon-Huang) [reply pkg/datastore/store.go:580]: track by https://github.com/mem9-ai/drive9/issues/110
